### ISSUE TITLE
Adding a simple script to read multiple Excel files?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#8521]
@Patowhiz we can import from multiple Excel sheets in the same worksbook.
If we have single sheets exported as csv files then we can read a directory of csv files (or sav files from SPSS).
But we can't (yet) read from a single (first) sheet of multiple Excel files.

I think you said that once loops were easy from the script window, then you could adapt the reading of a single sheet to enable this feature.

a) Well now we can handle loops in the script window.
b) And I am very keen to build up examples of scripts, for the R-Instat library, just as we have lots of datasets in the library.  It is becomeing a data and script library!
c) And this script would also be very useful for teaching purposes.
d) And I had examples where I could have used it on the recent workshop at CIMH.  Some participants had each station in a separate Excel file.

Do you have an example we could use.  Otherwise can you think of a better example than making one from the Cameroon data?  This is currently used for teaching data preparation, because each year is in a separate sheet.  We could make another version in its own directory, with each year in a single sheet.

The reason the script would be good for teaching is that it caters for a rare situation.  It is also a situation I don't want to encourage, i.e. lots of very simple Excel files.  It would be better to put them into a single book.  But I would like to explain the idea also that we want to avoid the "multiple copy-paste syndrome", whenever possible, i.e. when you repeat something routinely, then you would like the computer to do it, rather than you doing it!  

So if we have 3 Excel files, then we do it one-at-a-time.  But if we have 30, or 300, then we get the computer to do it!

By the way we could also add to this teaching point nicely to show how R permits us to do loops efficiently in many situations.  We could give a simple example, with lists, using the calculator.  That's already there.

**Comment by Patowhiz** [comment]
@rdstern very excited to hear that @lloyddewit has added loops support. I'll test this. 
@rdstern we certainly need to test this feature in the script window (to know it's limits) then think of how it can be added in dialogs that may require it.
I understand that we have a bug when importing data frame through the script window, the data frame is imported twice. That should be a quick fix in the `RunScript()` of `clsRLink`. But I cannot touch it because @lloyddewit is looking at improving the same subroutine in another PR.

**Comment by rdstern** [comment]
@lilyclements no, not yet.  And we sometimes could have used that, i.e. many Excel files, with a sheet on each.  Usually that can be quite old files, so sometimes xls.

I don't want to promote that, as it is not good practice.  One thought I had was for @Patowhiz to write a script for extending the use of Excel for one file, into the multiple one.  So, if just a few files, then just do them one at a time.  If lots - I once had 18 - then adapt the script.

Not high priority though.

**Comment by rdstern** [comment]
@rachelkg I first note here a small item for the Help on importing.  With the

**Comment by lilyclements** [comment]
@rdstern Yes, I think this should be possible in the script window using the same Excel import code that is generated by the dialog.

The idea would be to use the dialog to generate the import for one file, and then replace the single filename with a list of files and loop over them with `lapply()`.

For example, conceptually:

```r
files <- list.files(
  path = "path/to/files",
  pattern = "\\.xlsx$",
  full.names = TRUE
)

data_list <- lapply(files, function(file) {
  # the existing Excel import, using `file`
})
```

We could then add each resulting data frame to the data book, using the filenames to generate the data frame names.

So the Excel import specification would still come from the existing dialog, and the script would provide the extra step of applying that same specification to all the files in a directory.

E.g., with your zip file

1. Remove "~$METEO-CLIMSOFT-DOUALA- AOUT -1974" as this one causes issues (zipped? Not sure!)
2. run the following:

```r
# List the files that are in that directory:
files <- list.files(
  path = "C:/Users/lclem/Downloads/TCM-Douala_P30-1974/TCM-Douala_P30-1974",  # insert your file path here
  pattern = "\\.xlsx$",
  full.names = TRUE
)

lapply(files, function(f) {
  
  data <- rio::import_list(
    file = f,
    guess_max = Inf,
    which = c(1, 2, 3)
  )
  
  file_name <- tools::file_path_sans_ext(basename(f))
  
  data_book$import_data(
    data_tables = data,
    data_names = paste0(file_name, "_", c("RESUME", "CONTENU", "ELEMENTS"))
  )
})
```

We would want to fix the renaming of the new files, but is this along the right lines @rdstern?
